improver/cli/threshold.py

```diff
@@ -140,29 +140,32 @@ def process(
         thresholds = [np.float32(x) for x in threshold_values]
         fuzzy_bounds = None
 
-    result_no_collapse_coord = BasicThreshold(
+    each_threshold_func_list = []
+
+    if vicinity is not None:
+        # smooth thresholded occurrences over local vicinity
+        each_threshold_func_list.append(OccurrenceWithinVicinity(vicinity))
+
+    if collapse_coord is not None:
+        # Take a weighted mean across realizations with equal weights
+        each_threshold_func_list.append(
+            WeightAndBlend(collapse_coord, "linear", y0val=1.0, ynval=1.0)
+        )
+
+    result = BasicThreshold(
         thresholds,
         fuzzy_factor=fuzzy_factor,
         fuzzy_bounds=fuzzy_bounds,
         threshold_units=threshold_units,
         comparison_operator=comparison_operator,
+        each_threshold_func=each_threshold_func_list,
     )(cube)
 
     if vicinity is not None:
-        # smooth thresholded occurrences over local vicinity
-        result_no_collapse_coord = OccurrenceWithinVicinity(vicinity)(
-            result_no_collapse_coord
-        )
-        new_cube_name = in_vicinity_name_format(result_no_collapse_coord.name())
-        result_no_collapse_coord.rename(new_cube_name)
-
-    if collapse_coord is None:
-        return result_no_collapse_coord
+        result.rename(in_vicinity_name_format(result.name()))
```

Comment:
This shouldn't be here - it should be done in the `OccurrenceWithinVicinity` plugin, in the `process` method.  I've checked and it isn't there.  Please could you move it, or let me know and I can open a bug fix PR.

Reply:
I agree, but I would rather leave this to another PR. This needs a bit more refactor, because at the point where `OccurrenceWithinVicinity` is applied (inside the threshold loop), the cube name is not yet set to its final value (maybe it should?) and the regex inside `in_vicinity_name_format` doesn't match.

Reply:
Ah OK. Hadn't realised it wasn't trivial.  We can do this later.